File: Training_Assistant/TrainingAssistantFrontend/src/app/exercise/model/exercise.ts
import { MusclePart } from "src/app/muscle-part/model/muscle-part";
import { Type } from "./type.enum";

export class Exercise {
    id: number;
    name: string;
    burnedKcal: number;
    time: number;
    type: Type;
    muscleParts: MusclePart[];
  
    constructor(id: number, name: string, burnedKcal: number, time: number, type: Type, muscleParts: MusclePart[]) {
      this.id = id;
      this.name = name;
      this.burnedKcal = burnedKcal;
      this.time = time;
      this.type = type;
      this.muscleParts = muscleParts;
    }
  }

File: Training_Assistant/TraingAssistant/Models/Exercise.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

public enum TypeEnum
{
    aerobic,
    strength,
    flexibility,
    mobility,
    balance
};
namespace TraingAssistantDAL.Models
{
    public class Exercise
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double BurnedKcal { get; set; }
        public int Time { get; set; }
        public TypeEnum Type { get; set; }

        public ICollection<MusclePart> MuscleParts { get; set; } = new List<MusclePart>();
        //public List<MusclePart>? MuscleParts { get; set; }
        //public List<ExerciseMusclePart>? ExercisesMuscleParts { get; set; }
        //public List<TrainingExercise>? TrainingExercises { get; set; }
    }
}
